### PR TITLE
Integration with ODIM's services part #5

### PR DESCRIPTION
**Pull request by MaciejMis**
Changes:
- Implemented client for communication with Redfish entities.
- Implemented generic GET resource handler.
- Implemented Validate endpoint.

### CODE REVIEW

File: src/rest/validateHandler.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString(errorMessage)
+		return
+	}
+
+	if resp.StatusCode == http.StatusUnauthorized {
+		ctx.StatusCode(http.StatusUnauthorized)
+		ctx.WriteString("Authentication failed. Wrong username and/or password")
+		return
+	}
```
Comment (ehaligow) [inline src/rest/validateHandler.go:80]: maybe add new line after closing curly bracket?

File: src/utils/uriTranslator.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+func (u *uriTranslator) DmToRedfish(uri string) string {
+	return strings.Replace(uri, u.dm, u.redfish, -1)
+}
+
+func (u *uriTranslator) RedfishToDm(uri string) string {
+	return strings.Replace(uri, u.redfish, u.dm, -1)
+}
+
+var UriTranslator = &uriTranslator{
+	dm:      "ODIM",
```
Comment (ehaligow) [inline src/utils/uriTranslator.go:19]: This is confusing, can we rename it?

Comment (ehaligow) [inline src/utils/uriTranslator.go:19]: This is confusing, can we rename it?

File: src/utils/uriTranslator.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package utils
+
+import "strings"
+
+type uriTranslator struct {
```
Comment (ehaligow) [inline src/utils/uriTranslator.go:5]: The name is confusing, it isn't used for translation, but for replacement.

File: src/rest/validateHandler.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+type validateHandler struct {
+	cfg config.Config
+}
+
+func (v *validateHandler) handle(ctx iris.Context) {
+	if ctx.GetHeader("Authorization") == "" {
+		ctx.StatusCode(http.StatusUnauthorized)
+		logrus.Error("No valid authorization")
+		ctx.WriteString("No valid authorization")
+		return
```
Comment (ehaligow) [inline src/rest/validateHandler.go:23]: Below error messages are moved to a variable, as they are uses twice. Can we be consistent?

File: src/rest/getGenericResourceHandler.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+		ctx.StatusCode(http.StatusInternalServerError)
+		ctx.WriteString(errorMessage)
+		return
+	}
+
+	if response.StatusCode == http.StatusUnauthorized {
+		ctx.StatusCode(http.StatusUnauthorized)
+		ctx.WriteString("Authentication failed. Wrong username and/or password")
+		return
+	}
```
Comment (ehaligow) [inline src/rest/getGenericResourceHandler.go:60]: maybe add new line after closing curly bracket?

File: src/rest/validateHandler.go (hunk 1 above)
Comment (MaciejMis) [reply src/rest/validateHandler.go:80]: Fixed

File: src/rest/getGenericResourceHandler.go (hunk 5 above)
Comment (MaciejMis) [reply src/rest/getGenericResourceHandler.go:60]: Fixed

File: src/utils/uriTranslator.go (hunk 2 above)
Comment (MaciejMis) [reply src/utils/uriTranslator.go]: Renamed to UriConverter

File: src/utils/uriTranslator.go (hunk 3 above)
Comment (MaciejMis) [reply src/utils/uriTranslator.go]: Renamed to UriConverter

File: src/rest/validateHandler.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+		}
+	}
+
+	defer resp.Body.Close()
+	serviceRoot := &ServiceRoot{}
+	err = json.NewDecoder(resp.Body).Decode(serviceRoot)
+	if err != nil {
+		errorMessage := "Error while reading response body: " + err.Error()
+		logrus.Error(errorMessage)
+		ctx.StatusCode(http.StatusInternalServerError)
```
Comment (ehaligow) [inline src/rest/validateHandler.go:72]: this fragment of code is repeated 3 times in this func

Comment (MaciejMis) [reply src/rest/validateHandler.go:72]: It is already fixed.

**Pull request merged**